### ISSUE TITLE
Implement backup leader for signing requests

### ISSUE THREAD

**Issue by bowenwang1996** [Near-One/mpc#63]
In the current implementation, the hash of a signing request decides which node should be the leader. However, if that node is offline, then 1/N signing requests will time out. We should implement a mechanism that allows the next node to initiate the signing request after a timeout.

**Comment by robin-near** [comment]
@saketh-are this is complete right?

**Comment by saketh-are** [comment]
Yeah I think the intended effect is achieved, although not exactly as described in the issue:

https://github.com/Near-One/mpc/blob/71f8dfdf5c8efa4f1caa1b13f67d3b4959b54338/node/src/mpc_client.rs#L257-L263
